src/NuGet.Clients/NuGet.CommandLine/Common/Solution.cs

```diff
@@ -121,8 +122,19 @@ private void LoadSolutionWithMsbuild14(Assembly msbuildAssembly, string solution
             {
                 var projectType = project.ProjectType.ToString();
                 var isSolutionFolder = projectType.Equals("SolutionFolder", StringComparison.OrdinalIgnoreCase);
-                var relativePath = project.RelativePath.Replace('\\', Path.DirectorySeparatorChar);
-                projects.Add(new ProjectInSolution(relativePath, isSolutionFolder));
+                try
+                {
+                    var projectShouldBuild = projectShouldBuildMethod != null ? projectShouldBuildMethod.Invoke(solutionFile, new object[] { project.RelativePath }) : true;
+                    if (projectShouldBuild)
```

Comment:
One concern here, if we have:
- solution filter
- can't find the ProjectShouldBuild method

then we shouldn't blindly just build all projects in the solution. 

We should error out.

Reply:
Ok we can do that.
I don't know the code at all so I made it that way because for me LoadSolutionWithMsbuild14 can be called by older MsBuild version that doesn't support filtering and I don't want to broke it in this case.
But I can check if it's a slnf file and only do the check in this case, in this way if we pass an sln file to msbuild 14 or 15 it will not throw because the check will not be applied.

Reply:
Yeah that method is really about msbuild14 and later. Feel free to improve the name.

Reply:
Resolving this.